installer/Util/SystemConfig.php

```diff
@@ -0,0 +1,621 @@
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Installer\Util;
+
+use Exception;
+use OrangeHRM\Installer\Controller\Upgrader\Traits\UpgraderUtilityTrait;
+use Symfony\Component\Filesystem\Filesystem;
+
+class SystemConfig
+{
+    use UpgraderUtilityTrait;
+
+    public const PASSED = 1;
+    public const ACCEPTABLE = 2;
+    public const BLOCKER = 3;
+
+    public const ENGINE_INNODB = 'InnoDB';
+
+    public const STATE_DISABLED = 'DISABLED';
+    public const STATE_DEFAULT = 'DEFAULT';
+    public const STATE_YES = 'YES';
+    public const STATE_NO = 'NO';
+    public const STATE_ON = 'ON';
+
+    public const INSTALL_UTIL_MEMORY_NO_LIMIT = 0;
+    public const INSTALL_UTIL_MEMORY_UNLIMITED = 1;
+    public const INSTALL_UTIL_MEMORY_HARD_LIMIT_FAIL = 2;
+    public const INSTALL_UTIL_MEMORY_SOFT_LIMIT_FAIL = 3;
+    public const INSTALL_UTIL_MEMORY_OK = 4;
+
+    private bool $interruptContinue = false;
+    private ?Filesystem $filesystem = null;
+
+    public function __construct()
+    {
+        $this->filesystem = new Filesystem();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInterruptContinue(): bool
+    {
+        return $this->interruptContinue;
+    }
+
+    /**
+     * Category => Environment Check
+     */
+
+    /**
+     * PHP Version Check
+     * @return array
+     */
+    public function isPHPVersionCompatible(): array
+    {
+        if (version_compare(PHP_VERSION, Messages::PHP_MIN_VERSION) < 0) {
+            $this->interruptContinue = true;
+            return [
+                'message' => Messages::PHP_FAIL_MESSAGE . " Installed version is " . PHP_VERSION,
+                'status' => self::BLOCKER
+            ];
+        } else {
+            return [
+                'message' => Messages::PHP_OK_MESSAGE . " (ver " . PHP_VERSION . ")",
+                'status' => self::PASSED
+            ];
+        }
+    }
+
+    /**
+     * MYSQL Client Check
+     * @return array
+     */
+    public function isMySqlClientCompatible(): array
+    {
+        if (function_exists('mysqli_get_client_info')) {
+            $mysqlClient = mysqli_get_client_info();
```

Comment:
is there any way to get this via PDO

Reply:
PDO::getAttribute(PDO::ATTR_CLIENT_VERSION)
found this one. But it throws an exception